### PR TITLE
docs: describe strict mode & --no-strict flag

### PR DESCRIPTION
**Pull request by Lucas-C**
This behaviour is specified by this test : https://github.com/cucumber/cucumber-js/blob/master/features/strict_mode.feature

### CODE REVIEW

**Comment by charlierudolph** [comment]
Similar to comment on #1279, would like to have a header for `no-strict`

**Comment by Lucas-C** [comment]
PR updated accordingly

File: docs/cli.md [hunk 1]
```diff
@@ -79,8 +79,11 @@ By default, cucumber exits when the event loop drains. Use the `--exit` flag in
 
 Undefined steps snippets are printed in JavaScript using the callback interface by default.
 
-By default, cucumber is in `strict` mode, meaning it will fail if there are pending or undefined steps.
-This can be disabled by setting the `--no-strict` flag.
+## --no-strict
+
+disable _strict_ mode.
+
+By default, cucumber works in _strict_ mode, meaning it will fail if there are pending or undefined steps.
```
Comment (charlierudolph) [inline docs/cli.md]: Strict mode only effects pending steps. Cucumber-js always fails on undefined steps. (since v3.0.0)

Comment (Lucas-C) [reply docs/cli.md]: Fixed

**Pull request merged**